python_modules/dagster/dagster/_config/pythonic_config/__init__.py

```diff
@@ -226,12 +247,32 @@ def _as_config_dict(self) -> Mapping[str, Any]:
             field = self.__fields__.get(key)
             if field and value is None and not _is_pydantic_field_required(field):
                 continue
+
             if field:
                 output[field.alias] = value
             else:
                 output[key] = value
         return output
 
+    def _convert_to_config_dictionary_inner(
+        self,
```

Comment:
I think we can avoid the code duplication with _get_non_none_public_field_values by calling it in this implementation here.

```
public_fields = self._get_non_none_public_field_values()
return { k: _config_value_to_dict_representation(v) for k, v in public_fields.items() }
```